dss_client/test/test_dss_c_interface.c

```diff
@@ -77,10 +121,50 @@ int main(int argc, char* argv[]) {
 	ret = DeleteObject(c, (void*) obj_name, strlen(obj_name));
 	printf("Object testfile1 deleted\n");
 
+	// upload objects with a directory structure to test LIST
+	if (putRecursive(c, "/etc") < 0 ){
+		printf("Uploading directory failed.\n");
+		goto out;
+	}
+	//test the LIST
+	keys_list = (char*) malloc(sizeof(char) * max_key_len * GetPageSize());
```

Comment:
since sizeof(char) is 1, you can see if this is really required. This may be applicable at the top as well.